core/translate/schema.rs

```diff
@@ -767,11 +767,183 @@ pub fn translate_drop_table(
+    });
+    program.emit_insn(Insn::OpenRead {
+        cursor_id: sqlite_schema_cursor_id_1,
+        root_page: 1usize.into(),
+    });
+
+    let r8 = program.alloc_register();
+    let r9 = program.alloc_register();
+    let r10 = program.alloc_register();
+    let r11 = program.alloc_register();
+    let r12 = program.alloc_register();
+    let r13 = program.alloc_register();
+    let _r14 = program.alloc_register(); //  Unsure why this register is allocated but putting it in here to make comparison with SQLite easier
+    let r15 = program.alloc_register();
+
+    //  Loop to copy over row id's from the schema table for rows that have the same root page as the one that was moved
+    let end_metadata_label = program.allocate_label();
+    let metadata_loop = program.allocate_label();
+    program.emit_insn(Insn::Rewind {
+        cursor_id: sqlite_schema_cursor_id_1,
+        pc_if_empty: end_metadata_label,
+    });
+    program.preassign_label_to_next_insn(metadata_loop);
+    //  start loop on schema table
+    program.emit_insn(Insn::Column {
+        cursor_id: sqlite_schema_cursor_id_1,
+        column: 3,
+        dest: r13,
+    });
+    let next_label = program.allocate_label();
+    program.emit_insn(Insn::Ne {
+        lhs: r13,
+        rhs: r2_reg,
+        target_pc: next_label,
+        flags: CmpInsFlags::default(),
+        collation: program.curr_collation(),
+    });
+    program.emit_insn(Insn::RowId {
+        cursor_id: sqlite_schema_cursor_id_1,
+        dest: r7,
+    });
+    program.emit_insn(Insn::Insert {
+        cursor: ephemeral_cursor_id,
+        key_reg: r7,
+        record_reg: r6,
+        flag: 0,
+        table_name: "scratch_table".to_string(),
+    });
+
+    program.resolve_label(next_label, program.offset());
+    program.emit_insn(Insn::Next {
+        cursor_id: sqlite_schema_cursor_id_1,
+        pc_if_next: metadata_loop,
+    });
+    program.preassign_label_to_next_insn(end_metadata_label);
+    //  End loop to copy over row id's from the schema table for rows that have the same root page as the one that was moved
+
+    program.resolve_label(if_not_label, program.offset());
+
+    //  4. Open a write cursor to the schema table and re-insert the records placed in the ephemeral table but insert the correct root page now
+    program.emit_insn(Insn::OpenWrite {
+        cursor_id: sqlite_schema_cursor_id_1,
+        root_page: 1usize.into(),
+        name: SQLITE_TABLEID.to_string(),
+    });
 
-    //  4. TODO: Open a write cursor to the schema table and re-insert all triggers into the sqlite schema table from the ephemeral table and delete old trigger
-    //  Requires support via https://github.com/tursodatabase/limbo/pull/768
+    //  Loop to copy over row id's from the ephemeral table and then re-insert into the schema table with the correct root page
+    let end_metadata_label = program.allocate_label();
+    let metadata_loop = program.allocate_label();
```

Comment:
again this is not a "metadata loop", maybe `copy_temp_table_to_schema_loop`

Reply:
addressed by 780d0cb5a80438be1d9f9d31c3995ccfdbb21a20